### ISSUE TITLE
Allow for fixed validation dataset in Experiment

### ISSUE THREAD

**Issue by srcansiz** [fedbiomed/fedbiomed#505]
In GitLab by @sharkovsky on Apr 14, 2023, 15:56

As a data scientist, I want to be able to reuse the same validation dataset across rounds (instead of randomly sampling a new validation dataset each round).

**Comment by geokyr** [comment]
Adding the option to select a specified, different validation set during training would be a useful feature. I mean one that wouldn't be used in training and would be used just for validation and would be the same across rounds.

**Comment by vasileios-ektor-papoulias** [comment]
I would find it immensely useful to be granted the freedom to control the validation process as much as I control the training process. For instance, you could expose a test_data function that we override alongside test_step, similar to how training data and training_step work now.

**Comment by mvesin** [comment]
Update: discussion from SP38 preparation

–	first step: implement ratio of each dataset *dedicated* to testing, implement as default behaviour + do not remove current “validation” mode
–	second step: extension to more general scheme (specify per-dataset, specify policy of when to test, specify number of samples vs ratio see below)
–	also: security, implement minimum number of samples per dataset (at load time, at split time, eg applies to both the testing subset and training subset of dataset)

**Comment by ybouilla** [comment]
Feature implementation:
1. **Node side**
- update API
- update breakpoints
3. **Researcher side**
 - create a way to save dataset (list of indexes)

**Comment by ybouilla** [comment]
**Not covered in this issue**: I was wondering if displaying the validation and the training loss on the same graph could be interesting for the user, to proper evaluate underfitting / overfitting

**Comment by srcansiz** [comment]
> **Not covered in this issue**: I was wondering if displaying the validation and the training loss on the same graph could be interesting for the user, to proper evaluate underfitting / overfitting

Hi @ybouilla,

If we display both  the graph could be a bit complicated read. It is because currently we can stack feedback scalar values coming from different nodes,  and each scalar type has different plots. A plot with multiple nodes and multiple type of scalar can be complicated to read.

**Comment by ybouilla** [comment]
Agreed, but then you can isolated the losses /curves that you want to study (there is this option in tensorboard). I don't think it is that hard for the user to use.

**Comment by srcansiz** [comment]
> Agreed, but then you can isolated the losses /curves that you want to study (there is this option in tensorboard). I don't think it is that hard for the user to use.

yes you're right I didn't think about this feature!
